### ISSUE TITLE
Install fails on brewed poetry

### ISSUE THREAD

**Issue by tback** [volopivoshenko/poetry-plugin-dotenv#327]
### Description

Installation fails on python3.13 installed via brew because packages are managed by brew.

### Workarounds

Install poetry via pipx instead

### Operating System

macos 14

### poetry version

1.8.4

### poetry-plugin-dotenv version

2.4.0

### Example pyproject.toml

```TOML
[tool.poetry]
name = "tmp-emmnjsz5h2"
version = "0.1.0"
description = ""
authors = ["Till Backhaus <redacted>"]
readme = "README.md"
packages = [{include = "tmp"}]

[tool.poetry.dependencies]
python = "^3.13"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

### Runtime logs

```bash session
poetry self add poetry-plugin-dotenv                                                                                                  Using version ^2.4.0 for poetry-plugin-dotenv

Updating dependencies
Resolving dependencies... (0.1s)

Package operations: 8 installs, 25 updates, 0 removals

  - Downgrading certifi (2024.8.30 /opt/homebrew/lib/python3.13/site-packages -> 2023.7.22): Installing...
  - Downgrading charset-normalizer (3.4.0 -> 3.3.2): Failed

  CalledProcessError

  Command '['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'charset-normalizer', '-y']' returned non-zero exit status 1.

  at /opt/homebrew/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py:577 in run
       573│             # We don't call process.wait() as .__exit__ does that for us.
       574│             raise
       575│         retcode = process.poll()
       576│         if check and retcode:
    →  577│             raise CalledProcessError(retcode, process.args,
       578│                                      output=stdout, stderr=stderr)
       579│     return CompletedProcess(process.args, retcode, stdout, stderr)
       580│ 
       581│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'charset-normalizer', '-y'] errored with the following return code 1
  
  Output:
  Found existing installation: charset-normalizer 3.4.0
  error: uninstall-no-record-file
  
  × Cannot uninstall charset-normalizer 3.4.0
  ╰─> The package's contents are unknown: no RECORD file was found for charset-normalizer.
  
  hint: The package was installed by brew. You should check if it can uninstall the package.
  

  at /opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/poetry/utils/env/base_env.py:342 in _run
      338│                 output = subprocess.check_output(
      339│                     cmd, stderr=stderr, env=env, text=True, **kwargs
      340│                 )
      341│         except CalledProcessError as e:
    → 342│             raise EnvCommandError(e)
      343│ 
      344│         return output
      345│ 
      346│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

Cannot install charset-normalizer.

  - Downgrading idna (3.10 -> 3.4): Failed

  CalledProcessError

  Command '['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'idna', '-y']' returned non-zero exit status 1.

  at /opt/homebrew/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py:577 in run
       573│             # We don't call process.wait() as .__exit__ does that for us.
       574│             raise
       575│         retcode = process.poll()
       576│         if check and retcode:
    →  577│             raise CalledProcessError(retcode, process.args,
       578│                                      output=stdout, stderr=stderr)
       579│     return CompletedProcess(process.args, retcode, stdout, stderr)
       580│ 
       581│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'idna', '-y'] errored with the following return code 1
  
  Output:
  Found existing installation: idna 3.10
  error: uninstall-no-record-file
  
  × Cannot uninstall idna 3.10
  ╰─> The package's contents are unknown: no RECORD file was found for idna.
  
  hint: The package was installed by brew. You should check if it can uninstall the package.
  

  at /opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/poetry/utils/env/base_env.py:342 in _run
      338│                 output = subprocess.check_output(
      339│                     cmd, stderr=stderr, env=env, text=True, **kwargs
      340│                 )
      341│         except CalledProcessError as e:
    → 342│             raise EnvCommandError(e)
      343│ 
      344│         return output
      345│ 
      346│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

Cannot install idna.

  - Downgrading more-itertools (10.5.0 -> 10.1.0): Failed
  - Downgrading charset-normalizer (3.4.0 -> 3.3.2): Failed

  CalledProcessError

  Command '['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'charset-normalizer', '-y']' returned non-zero exit status 1.

  at /opt/homebrew/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py:577 in run
       573│             # We don't call process.wait() as .__exit__ does that for us.
       574│             raise
       575│         retcode = process.poll()
       576│         if check and retcode:
    →  577│             raise CalledProcessError(retcode, process.args,
       578│                                      output=stdout, stderr=stderr)
       579│     return CompletedProcess(process.args, retcode, stdout, stderr)
       580│ 
       581│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'charset-normalizer', '-y'] errored with the following return code 1
  
  Output:
  Found existing installation: charset-normalizer 3.4.0
  error: uninstall-no-record-file
  
  × Cannot uninstall charset-normalizer 3.4.0
  ╰─> The package's contents are unknown: no RECORD file was found for charset-normalizer.
  
  hint: The package was installed by brew. You should check if it can uninstall the package.
  

  at /opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/poetry/utils/env/base_env.py:342 in _run
      338│                 output = subprocess.check_output(
      339│                     cmd, stderr=stderr, env=env, text=True, **kwargs
      340│                 )
      341│         except CalledProcessError as e:
    → 342│             raise EnvCommandError(e)
      343│ 
      344│         return output
      345│ 
      346│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

Cannot install charset-normalizer.

  - Downgrading idna (3.10 -> 3.4): Failed

  CalledProcessError

  Command '['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'idna', '-y']' returned non-zero exit status 1.

  at /opt/homebrew/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py:577 in run
       573│             # We don't call process.wait() as .__exit__ does that for us.
       574│             raise
       575│         retcode = process.poll()
       576│         if check and retcode:
    →  577│             raise CalledProcessError(retcode, process.args,
       578│                                      output=stdout, stderr=stderr)
       579│     return CompletedProcess(process.args, retcode, stdout, stderr)
       580│ 
       581│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'idna', '-y'] errored with the following return code 1
  
  Output:
  Found existing installation: idna 3.10
  error: uninstall-no-record-file
  
  × Cannot uninstall idna 3.10
  ╰─> The package's contents are unknown: no RECORD file was found for idna.
  
  hint: The package was installed by brew. You should check if it can uninstall the package.
  

  at /opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/poetry/utils/env/base_env.py:342 in _run
      338│                 output = subprocess.check_output(
      339│                     cmd, stderr=stderr, env=env, text=True, **kwargs
      340│                 )
      341│         except CalledProcessError as e:
    → 342│             raise EnvCommandError(e)
      343│ 
      344│         return output
      345│ 
      346│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

Cannot install idna.

  - Downgrading more-itertools (10.5.0 -> 10.1.0): Failed
  - Downgrading certifi (2024.8.30 /opt/homebrew/lib/python3.13/site-packages -> 2023.7.22)
  - Downgrading charset-normalizer (3.4.0 -> 3.3.2): Failed

  CalledProcessError

  Command '['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'charset-normalizer', '-y']' returned non-zero exit status 1.

  at /opt/homebrew/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py:577 in run
       573│             # We don't call process.wait() as .__exit__ does that for us.
       574│             raise
       575│         retcode = process.poll()
       576│         if check and retcode:
    →  577│             raise CalledProcessError(retcode, process.args,
       578│                                      output=stdout, stderr=stderr)
       579│     return CompletedProcess(process.args, retcode, stdout, stderr)
       580│ 
       581│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'charset-normalizer', '-y'] errored with the following return code 1
  
  Output:
  Found existing installation: charset-normalizer 3.4.0
  error: uninstall-no-record-file
  
  × Cannot uninstall charset-normalizer 3.4.0
  ╰─> The package's contents are unknown: no RECORD file was found for charset-normalizer.
  
  hint: The package was installed by brew. You should check if it can uninstall the package.
  

  at /opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/poetry/utils/env/base_env.py:342 in _run
      338│                 output = subprocess.check_output(
      339│                     cmd, stderr=stderr, env=env, text=True, **kwargs
      340│                 )
      341│         except CalledProcessError as e:
    → 342│             raise EnvCommandError(e)
      343│ 
      344│         return output
      345│ 
      346│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

Cannot install charset-normalizer.

  - Downgrading idna (3.10 -> 3.4): Failed

  CalledProcessError

  Command '['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'idna', '-y']' returned non-zero exit status 1.

  at /opt/homebrew/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py:577 in run
       573│             # We don't call process.wait() as .__exit__ does that for us.
       574│             raise
       575│         retcode = process.poll()
       576│         if check and retcode:
    →  577│             raise CalledProcessError(retcode, process.args,
       578│                                      output=stdout, stderr=stderr)
       579│     return CompletedProcess(process.args, retcode, stdout, stderr)
       580│ 
       581│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'idna', '-y'] errored with the following return code 1
  
  Output:
  Found existing installation: idna 3.10
  error: uninstall-no-record-file
  
  × Cannot uninstall idna 3.10
  ╰─> The package's contents are unknown: no RECORD file was found for idna.
  
  hint: The package was installed by brew. You should check if it can uninstall the package.
  

  at /opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/poetry/utils/env/base_env.py:342 in _run
      338│                 output = subprocess.check_output(
      339│                     cmd, stderr=stderr, env=env, text=True, **kwargs
      340│                 )
      341│         except CalledProcessError as e:
    → 342│             raise EnvCommandError(e)
      343│ 
      344│         return output
      345│ 
      346│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

Cannot install idna.

  - Downgrading more-itertools (10.5.0 -> 10.1.0): Failed

  CalledProcessError

  Command '['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'more-itertools', '-y']' returned non-zero exit status 1.

  at /opt/homebrew/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py:577 in run
       573│             # We don't call process.wait() as .__exit__ does that for us.
       574│             raise
       575│         retcode = process.poll()
       576│         if check and retcode:
    →  577│             raise CalledProcessError(retcode, process.args,
       578│                                      output=stdout, stderr=stderr)
       579│     return CompletedProcess(process.args, retcode, stdout, stderr)
       580│ 
       581│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'more-itertools', '-y'] errored with the following return code 1
  
  Output:
  Found existing installation: more-itertools 10.5.0
  error: uninstall-no-record-file
  
  × Cannot uninstall more-itertools 10.5.0
  ╰─> The package's contents are unknown: no RECORD file was found for more-itertools.
  
  hint: The package was installed by brew. You should check if it can uninstall the package.
  

  at /opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/poetry/utils/env/base_env.py:342 in _run
      338│                 output = subprocess.check_output(
      339│                     cmd, stderr=stderr, env=env, text=True, **kwargs
      340│                 )
      341│         except CalledProcessError as e:
    → 342│             raise EnvCommandError(e)
      343│ 
      344│         return output
      345│ 
      346│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

Cannot install more-itertools.

  - Downgrading pycparser (2.22 /opt/homebrew/lib/python3.13/site-packages -> 2.21)
  - Downgrading urllib3 (2.2.3 -> 2.1.0): Failed

  CalledProcessError

  Command '['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'urllib3', '-y']' returned non-zero exit status 1.

  at /opt/homebrew/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py:577 in run
       573│             # We don't call process.wait() as .__exit__ does that for us.
       574│             raise
       575│         retcode = process.poll()
       576│         if check and retcode:
    →  577│             raise CalledProcessError(retcode, process.args,
       578│                                      output=stdout, stderr=stderr)
       579│     return CompletedProcess(process.args, retcode, stdout, stderr)
       580│ 
       581│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/opt/homebrew/Cellar/poetry/1.8.4/libexec/bin/python', '/opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl/pip', 'uninstall', 'urllib3', '-y'] errored with the following return code 1
  
  Output:
  Found existing installation: urllib3 2.2.3
  error: uninstall-no-record-file
  
  × Cannot uninstall urllib3 2.2.3
  ╰─> The package's contents are unknown: no RECORD file was found for urllib3.
  
  hint: The package was installed by brew. You should check if it can uninstall the package.
  

  at /opt/homebrew/Cellar/poetry/1.8.4/libexec/lib/python3.13/site-packages/poetry/utils/env/base_env.py:342 in _run
      338│                 output = subprocess.check_output(
      339│                     cmd, stderr=stderr, env=env, text=True, **kwargs
      340│                 )
      341│         except CalledProcessError as e:
    → 342│             raise EnvCommandError(e)
      343│ 
      344│         return output
      345│ 
      346│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

Cannot install urllib3.
```

**Comment by pivoshenko** [comment]
Hi @tback! Thanks for submitting the issue, I really appreciate the feedback

I wanted to apologize that I didn't have a chance to have a look into it, firstly was super busy with work and now I am ill, but once I will recover I will look into it!

Thanks again!

**Comment by pivoshenko** [comment]
Hi @tback! First of all, I want to apologize for taking so much time to look into this issue (it was a crazy couple of weeks with a lot of stuff going around)!

So I checked and the main problem is how poetry treats its core dependencies, unfortunately for self commands there is no way to set no update mode.

But there is some trick that helped me to solve your case, prior to running the plugin installation command (**outside of venv**) run the next command:

```bash
python3 -m pip install --break-system-packages -U --no-cache-dir poetry cleo rapidfuzz importlib_metadata zipp crashtest
```

make sure that `python3` points to the Python3.13

and then run:

```bash
poetry self add poetry-plugin-dotenv@latest
```

Thanks, let me know if it helped, and if not just re-open the issue! I tested it out in the Docker container.
